### PR TITLE
Feature/add vpc config

### PR DESCRIPTION
**Pull request by ikopas**
*Issue #, if available:* 15

*Description of changes:* Added support for running the AWS CodeBuild Project within a VPC and expanded the list of supported ECR repos containing pre-built Docker images maintained by AWS. Updated the README with details regarding how to use VPC arguments and listed the changes required to the access policy on the execution role.

By submitting this pull request, I confirm that you can use, modify, copy, and redistribute this contribution, under the terms of your choice.

### CODE REVIEW

File: README.md [hunk 1]
```diff
@@ -20,12 +20,18 @@ Any additional arguments supported with `docker build` are supported
...
 ```
 
-By default, the image will be pushed to a repository `sagemakerstudio` with the tag `latest`, and use the Studio App's execution role and the default SageMaker Python SDK S3 bucket
+By default, the CodeBuild project will not run within a VPC, the image will be pushed to a repository `sagemakerstudio` with the tag `latest`, and use the Studio App's execution role and the default SageMaker Python SDK S3 bucket
 
 These can be overridden with the relevant CLI options.
 
 ```bash
-sm-docker build . --repository mynewrepo:1.0 --role MyRoleName 
+sm-docker build . --repository mynewrepo:1.0 --role MyRoleName --bucket MyBucketName --vpc-id MyVpcId --subnets MySubnetId1,MySubnetId2 --security-groups MySecurityGroup1,MySecurityGroup2
```
Comment (jaipreet-s) [inline README.md:28]:
```suggestion
# Basic usage
sm-docker build .

# Specify a custom ECR repository to push to and custom role
sm-docker build . --repository mynewrepo:1.0 --role MyRoleName

# Connect the CodeBuild job to a VPC
sm-docker build . --repository mynewrepo:1.0 --role MyRoleName --bucket MyBucketName --vpc-id MyVpcId --subnets MySubnetId1,MySubnetId2 --security-groups MySecurityGroup1,MySecurityGroup2
```

File: README.md [hunk 2]
```diff
@@ -20,12 +20,18 @@ Any additional arguments supported with `docker build` are supported
...
+``` 
+
+For instance, please see the sample command with additional argument listed below:
+
+```bash
+sm-docker build . --repository mynewrepo:1.0 --role SampleDockerBuildRole --bucket sagemaker-us-east-1-326543455535 --vpc-id vpc-0c70e76ef1c603b94 --subnets subnet-0d984f080338960bb,subnet-0ac3e96808c8092f2 --security-groups sg-0d31b4042f2902cd0
```
Comment (jaipreet-s) [inline README.md:34]: nit-pick: I believe you can just add this in the example command on line #28 itself.

File: sagemaker_studio_image_build/cli.py [hunk 3]
```diff
@@ -85,6 +107,18 @@ def main():
...
+        "--vpc-id",
+        help="The Id of the VPC that will host the CodeBuild Project (such as vpc-05c09f91d48831c8c).",
+    )
+    build_parser.add_argument(
+        "--subnets",
+        help="The comma-separated list of subnet ids for the CodeBuild Project (such as subnet-0b31f1863e9d31a67)",
+    )
+    build_parser.add_argument(
+        "--security-groups",
+        help="The comma-separated list of security group ids for the CodeBuild Project (such as sg-0ce4ec0d0414d2ddc).",
```
Comment (jaipreet-s) [inline sagemaker_studio_image_build/cli.py:120]: In the VPC parameter, we're calling it `vpc-id` explicitly whereas in the subnets and security groups we're just calling them `subnets/security-groups` though we're expecting the user to pass IDs. I recommend we make the terminology here consistent

* --vpc, --subnets, --security-groups
OR
* --vpc-id, --subnet-ids, --security-groups-ids

My preference is the latter as it makes it clear to the user that the IDs are expected and not any other attribute of the VPC/Subnet/SecurityGroups like name

File: sagemaker_studio_image_build/data/buildspec.template.yml [hunk 4]
```diff
@@ -9,6 +9,9 @@ phases:
       - $(aws ecr get-login --no-include-email --region $AWS_DEFAULT_REGION --registry-ids 217643126080)
       - $(aws ecr get-login --no-include-email --region $AWS_DEFAULT_REGION --registry-ids 727897471807)
       - $(aws ecr get-login --no-include-email --region $AWS_DEFAULT_REGION --registry-ids 626614931356)
+      - $(aws ecr get-login --no-include-email --region $AWS_DEFAULT_REGION --registry-ids 683313688378)
+      - $(aws ecr get-login --no-include-email --region $AWS_DEFAULT_REGION --registry-ids 520713654638)
+      - $(aws ecr get-login --no-include-email --region $AWS_DEFAULT_REGION --registry-ids 462105765813)
```
Comment (jaipreet-s) [inline sagemaker_studio_image_build/data/buildspec.template.yml:14]: Please add a comment on what Deep Learning Container or Algorithm container these correspond to.

**Review by jaipreet-s — CHANGES_REQUESTED** [review]
Looks good overall! A few comments that we should address before merging and releasing.

File: README.md (hunk 1 above)
Comment (jaipreet-s) [reply README.md:28]: My recommendation is that we keep the basic example as-is,  but add more "advanced" usage examples like using VPC.

File: sagemaker_studio_image_build/cli.py (hunk 3 above)
Comment (ikopas) [reply sagemaker_studio_image_build/cli.py]: Standardized the input arguments using the latter of the two options

File: README.md (hunk 1 above)
Comment (ikopas) [reply README.md]: Addressed as part of the change recommended on the second comment below, where the suggestion was made to use the extended example with sample values in place of the earlier sample example

File: sagemaker_studio_image_build/data/buildspec.template.yml (hunk 4 above)
Comment (ikopas) [reply sagemaker_studio_image_build/data/buildspec.template.yml:14]: Please see the details below:
- 683313688378 - [Prebuilt Amazon SageMaker Docker Images for Scikit-learn and Spark ML in us-east-1](https://docs.aws.amazon.com/sagemaker/latest/dg/pre-built-docker-containers-scikit-learn-spark.html)
- 520713654638 & 462105765813 - [Amazon SageMaker RL Containers](https://github.com/aws/sagemaker-rl-container)

**Review by jaipreet-s — APPROVED** [review]

**Pull request merged**